diff -Naur dropbear-2022.83/svr-chansession.c dropbear-2022.83-patch/svr-chansession.c
--- dropbear-2022.83/svr-chansession.c	2022-11-14 09:30:00.000000000 -0500
+++ dropbear-2022.83-patch/svr-chansession.c	2023-04-21 17:07:39.133601200 -0400
@@ -1046,6 +1046,7 @@
 	}
 #endif
 
+#if !DROPBEAR_SVR_NO_CHDIR
 	/* change directory */
 	if (chdir(ses.authstate.pw_dir) < 0) {
 		int e = errno;
@@ -1054,6 +1055,7 @@
 		}
 		fprintf(stderr, "Failed chdir '%s': %s\n", ses.authstate.pw_dir, strerror(e));
 	}
+#endif
 
 
 #if DROPBEAR_X11FWD
